.github/workflows/ci.yml

```
@@ -0,0 +1,15 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@v3.0.1
```

For [external actions](https://github.com/ACCESS-NRI/dev-docs/wiki/Continuous-integration-and-Deployment#actions-versioning), we should use the commit reference rather than the tag reference. This is because we can't be certain of external actions stability, or their creators security policies. 
```suggestion
      - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd  # v3.0.1
```

---

Makes sense!!